src/en/dynasty/src/eu/kanade/tachiyomi/extension/en/dynasty/Dynasty.kt

```diff
@@ -0,0 +1,698 @@
+            )
+            .fragment(COVER_URL_FRAGMENT)
+            .build()
+            .toString()
+    }
+
+    private fun buildChapterCoverFetchUrl(permalink: String): String {
+        return HttpUrl.Builder().apply {
+            scheme("http")
+            host(COVER_FETCH_HOST)
+            addQueryParameter("permalink", permalink)
+        }.build().toString()
+    }
+
+    private fun fetchCoverUrlInterceptor(chain: Interceptor.Chain): Response {
+        val request = chain.request()
+
+        if (request.url.host != COVER_FETCH_HOST) {
+            return chain.proceed(request)
+        }
+
+        val permalink = request.url.queryParameter("permalink")!!
+
+        val chapterUrl = baseUrl.toHttpUrl().newBuilder().apply {
+            addPathSegment("chapters")
+            addPathSegments("$permalink.json")
+        }.build()
+
+        val page = client.newCall(GET(chapterUrl, headers)).execute()
+            .parseAs<ChapterResponse>()
+            .pages.first()
+
+        val url = buildCoverUrl(page.url)
+
+        val newRequest = request.newBuilder()
+            .url(url)
+            .build()
+
+        return chain.proceed(newRequest)
+    }
+
+    private fun coverInterceptor(chain: Interceptor.Chain): Response {
+        val request = chain.request()
+
+        return if (request.url.fragment == COVER_URL_FRAGMENT) {
+            coverClient.newCall(request).execute()
+        } else {
+            chain.proceed(request)
+        }
+    }
+
+    private fun String.permalinkToTitle(): String {
+        val result = StringBuilder(length)
+        var capitalize = true
+        for (char in this) {
+            result.append(
+                if (char == '_') {
+                    ' '
+                } else if (capitalize) {
+                    char.uppercase()
+                } else {
+                    char.lowercase()
+                },
+            )
+
+            capitalize = char == '_'
+        }
+
+        return result.toString()
+    }
+
+    override fun imageUrlParse(response: Response) =
+        throw UnsupportedOperationException()
+    override fun latestUpdatesRequest(page: Int) =
+        throw UnsupportedOperationException()
+    override fun latestUpdatesParse(response: Response) =
+        throw UnsupportedOperationException()
+    override fun searchMangaParse(response: Response) =
+        throw UnsupportedOperationException()
+}
+
+private const val COVER_FETCH_HOST = "keiyoushi-chapter-cover"
+private const val COVER_URL_FRAGMENT = "thumbnail"
+private val CHAPTER_SLUG_REGEX = Regex("""(.*?)_(ch[0-9_]+|volume_[0-9_\w]+)""")
+private val UNICODE_REGEX = Regex("\\\\u([0-9A-Fa-f]{4})")
+private val dateFormat = SimpleDateFormat("yyyy-MM-dd", Locale.ENGLISH)
+private const val CHAPTER_FETCH_LIMIT_PREF = "chapterFetchLimit"
+private val CHAPTER_FETCH_LIMITS = arrayOf("2", "5", "10", "all")
+private val VALID_URL_REGEX = Regex("""^/(series|anthologies|doujins|issues|chapters)/.*""")
```

Comment:
Don't need last part if not using 'group 0'.

```suggestion
private val VALID_URL_REGEX = Regex("""^/(series|anthologies|doujins|issues|chapters)/""")
```

Reply:
changed check to use httpurl instead